### PR TITLE
Updated README

### PR DESCRIPTION
**Pull request by AnthonyOliveri**
- Added lots of content, including usage examples
- Used the same format as Swift [BMSCore](https://github.com/ibm-bluemix-mobile-services/bms-clientsdk-swift-core)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,114 +1,235 @@
-IBM Bluemix Mobile Services - Client SDK Android Core
+BMSCore
 ===
 
 [![Build Status](https://travis-ci.org/ibm-bluemix-mobile-services/bms-clientsdk-android-core.svg?branch=master)](https://travis-ci.org/ibm-bluemix-mobile-services/bms-clientsdk-android-core)
-[![Build Status](https://travis-ci.org/ibm-bluemix-mobile-services/bms-clientsdk-android-core.svg?branch=development)](https://travis-ci.org/ibm-bluemix-mobile-services/bms-clientsdk-android-core)
-[![Maven Central](https://maven-badges.herokuapp.com/maven-central/com.ibm.mobilefirstplatform.clientsdk.android/core/badge.svg)](https://maven-badges.herokuapp.com/maven-central/com.ibm.mobilefirstplatform.clientsdk.android/core)
```
Comment (dgonz7) [inline README.md:6]: Leave this here, this is standard for all Java github repos.

File: README.md [hunk 2]
```diff
@@ -1,114 +1,235 @@
...
 [![Codacy Badge](https://api.codacy.com/project/badge/Grade/33762c419c1a4743a0348c93686acb1c)](https://www.codacy.com/app/ibm-bluemix-mobile-services/bms-clientsdk-android-core?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=ibm-bluemix-mobile-services/bms-clientsdk-android-core&amp;utm_campaign=Badge_Grade)
 [![Coverage Status](https://coveralls.io/repos/github/ibm-bluemix-mobile-services/bms-clientsdk-android-core/badge.svg?branch=development)](https://coveralls.io/github/ibm-bluemix-mobile-services/bms-clientsdk-android-core?branch=development)
-[![javadoc.io](https://javadoc-emblem.rhcloud.com/doc/com.ibm.mobilefirstplatform.clientsdk.android/core/badge.svg)](http://www.javadoc.io/doc/com.ibm.mobilefirstplatform.clientsdk.android/core)
```
Comment (dgonz7) [inline README.md:9]: Leave this here, it is the only direct link to the Javadoc.

**Review by dgonz7 — CHANGES_REQUESTED** [review]
Aside from those two minor changes I noted, everything else is fine.

File: README.md (hunk 1 above)
Comment (AnthonyOliveri) [reply README.md:6]: Are you referring to the Build Status, or Maven Central badge? The Maven Central badge is still there, just moved to a different spot. I think it's confusing to have 2 badges for Build Status, especially when one is passing and one is failing. Most Github repos have just one build status.

File: README.md (hunk 2 above)
Comment (AnthonyOliveri) [reply README.md:9]: I added another link to the Javadoc in the Example Usage section. Do you still want to keep this Javadoc badge?

File: README.md (hunk 1 above)
Comment (AnthonyOliveri) [reply README.md:6]: Another note: I have a [task](https://github.ibm.com/arf/planning-devex/issues/159) to update the README badges for all SDKs (Swift, Android, and Cordova) to be consistent. So whatever we decide will need to be done to all other SDKs for all platforms (except for platform-dependent badges like Maven Central, of course).

Comment (dgonz7) [reply README.md:6]: Yup, meant Maven Central one, didn't see you moved it.

File: README.md (hunk 2 above)
Comment (dgonz7) [reply README.md:9]: Nope that's fine, as long as it's somewhere.

**Review by dgonz7 — APPROVED** [review]

**Pull request merged**